test/JDBC/expected/BABEL-3556-vu-prepare.out

```diff
@@ -0,0 +1,266 @@
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: value "922337203685478" is out of range for type money)~~
+
+
+INSERT INTO babel_3556_vu_prepare_t3 VALUES(₧-922337203685478);
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: value "-922337203685478" is out of range for type money)~~
+
+
+INSERT INTO babel_3556_vu_prepare_t3 VALUES(1232422322334223423121);
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: value "1232422322334223423121" is out of range for type fixeddecimal)~~
+
+
+-- To test out of range values for smallmoney
+CREATE TABLE babel_3556_vu_prepare_t5(a smallmoney);
+GO
+
+CREATE TABLE babel_3556_vu_prepare_t6(a smallmoney);
+GO
+
+-- decimal
+SELECT CAST($214748.3648 AS smallmoney);
+GO
+~~START~~
+smallmoney
+~~ERROR (Code: 220)~~
+
+~~ERROR (Message: value for domain smallmoney violates check constraint "smallmoney_check")~~
+
+
+SELECT CAST(₧-214748.3649 AS smallmoney);
+GO
+~~START~~
+smallmoney
+~~ERROR (Code: 220)~~
+
+~~ERROR (Message: value for domain smallmoney violates check constraint "smallmoney_check")~~
+
+
+INSERT INTO babel_3556_vu_prepare_t5 VALUES($214748.3648);
+GO
+~~ERROR (Code: 220)~~
+
+~~ERROR (Message: value for domain smallmoney violates check constraint "smallmoney_check")~~
+
+
+INSERT INTO babel_3556_vu_prepare_t5 VALUES(₧-214748.3649);
+GO
+~~ERROR (Code: 220)~~
+
+~~ERROR (Message: value for domain smallmoney violates check constraint "smallmoney_check")~~
+
+
+-- integer
+SELECT CAST($214749 AS smallmoney);
+GO
+~~START~~
+smallmoney
+~~ERROR (Code: 220)~~
+
+~~ERROR (Message: value for domain smallmoney violates check constraint "smallmoney_check")~~
+
+
+SELECT CAST(₧-214749 AS smallmoney);
+GO
+~~START~~
+smallmoney
+~~ERROR (Code: 220)~~
+
+~~ERROR (Message: value for domain smallmoney violates check constraint "smallmoney_check")~~
+
+
+SELECT CAST($1232422322334223423121 AS smallmoney);
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: value "1232422322334223423121" is out of range for type fixeddecimal)~~
+
+
+SELECT CAST(₧-1232422322334223423121 AS smallmoney);
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: value "-1232422322334223423121" is out of range for type fixeddecimal)~~
+
+
+INSERT INTO babel_3556_vu_prepare_t5 VALUES($214749);
+GO
+~~ERROR (Code: 220)~~
+
+~~ERROR (Message: value for domain smallmoney violates check constraint "smallmoney_check")~~
+
+
+INSERT INTO babel_3556_vu_prepare_t5 VALUES(₧-214749 );
+GO
+~~ERROR (Code: 220)~~
+
+~~ERROR (Message: value for domain smallmoney violates check constraint "smallmoney_check")~~
+
+
+INSERT INTO babel_3556_vu_prepare_t5 VALUES($1232422322334223423121);
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: value "1232422322334223423121" is out of range for type fixeddecimal)~~
+
+
+INSERT INTO babel_3556_vu_prepare_t5 VALUES(₧-1232422322334223423121);
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: value "-1232422322334223423121" is out of range for type fixeddecimal)~~
+
+
+-- cast inside view
```

Comment:
Please also add tests for func/proc where its input/output parameters are of money/smallmoney types.

Reply:
Added